lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new' && module.state !== 'errored') {
```

Comment:
Yeah I _strongly_ prefer we consistently don't allow redefines in all scenarios by defining clear boundaries. 

Also once this is sorted out, can you also update the README with this behavior? Might help developers debugging things to understand the expected behavior.